framework/TApplication.php

```diff
@@ -698,6 +699,60 @@ public function getModules()
 		return $this->_modules;
 	}
 
+	/**
+	 * Returns a list of application modules of a specific class.
+	 * Modules that have not been loaded yet are loaded.
+	 * @param string $type class name of the modules to look for.
+	 * @param bool $strict should just the class be found or subclasses of the class
+	 * @param bool $first return the first module found, default false.
+	 * @return array|object list of loaded application module of a specific class
+	 * or the first object of a specific class
+	 * @since 4.2.0
+	 */
+	public function getModulesByType($type, $strict = false, $first = false)
```

Comment:
Imho this method makes more things that what it should, making it a bit too complex.
I'd like it to return a list of modules ids, delegating to the caller the burden of calling `Prado::getApplication->getModule($id)` if it wants to use a module.
Since the caller would get an array of strings, we could also remove the $first parameter to make the method signature more strict on return types.
Please see the following comments.